### PR TITLE
const-prop #6780: Allow symbolic propagation for rvals in lhs of assignments

### LINKED ISSUE #6780: `symbolic_propagation` does not work properly for Golang code
**Describe the bug**
The symbolic_propagation does not work where the `...` is used for the Golang code.

**To Reproduce**
- https://semgrep.dev/s/Lkvo

**Expected behavior**
Match also:
```go
    x := http.DefaultClient
    y := x

    y.Timeout = 4 * time.Millisecond
```
and
```go
    x := http.DefaultClient
    x.Timeout = 3 * time.Millisecond
```

For the pattern without the `...` it works properly: https://semgrep.dev/s/8d5E

### PR DESCRIPTION
**Pull request by ihji**
Closes #6780 

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: tests/patterns/python/cp_label.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     #ERROR:
     "foo"
 
-#OK:
+#ERROR: a is foo by constant propagation
```
Comment (IagoAbal) [inline tests/patterns/python/cp_label.py]: Given the nature of the test I would have changed this to:
```python
#OK:
a = "bar"
```

File: src/il/IL_helpers.ml [hunk 2]
```diff
@@ -30,6 +30,8 @@ let exp_of_arg arg =
 
 let rexps_of_instr x =
   match x.i with
```
Comment (IagoAbal) [inline src/il/IL_helpers.ml:32]: You're missing a changelog in this PR.

File: tests/patterns/python/cp_label.py (hunk 1 above)
Comment (ihji) [reply tests/patterns/python/cp_label.py]: Done.

File: src/il/IL_helpers.ml (hunk 2 above)
Comment (ihji) [reply src/il/IL_helpers.ml:32]: Done.

**Review by IagoAbal — APPROVED** [review]

**Pull request merged**